packages/rum-core/src/domain/rumEventsCollection/action/clickChain.ts

```diff
@@ -0,0 +1,100 @@
+import type { TimeStamp } from '@datadog/browser-core'
+import { monitor, ONE_SECOND } from '@datadog/browser-core'
+
+export interface ClickChain<Click extends BaseClick> {
+  tryAppend: (click: Click) => null | ClickReference
+  stop: () => void
+}
+
+export interface ClickReference {
+  markAsComplete: () => void
+}
+
+export interface BaseClick {
+  event: MouseEvent
+  timeStamp: TimeStamp
+}
+
+export const CLICK_CHAIN_WINDOW_SIZE = 3
+export const CLICK_CHAIN_MAX_DURATION_WINDOW = ONE_SECOND
+export const CLICK_CHAIN_MAX_DISTANCE_WINDOW = 100
+
+const enum ClickChainStatus {
+  WaitingForMoreClicks,
+  WaitingForClicksToComplete,
+  Flushed,
+}
+
+export function createClickChain<Click extends BaseClick>(onFlush: (clicks: Click[]) => void): ClickChain<Click> {
```

Comment:
Like the idea of a click chain :) 
The more I look at it the more I think it's tightly related to rage click detection. 
So to me the naming should be more specific. Also I'm wondering if we could move `flushClickChain` inside of it.
Wdyt?